aptos/contract/aux/sources/stake.move

```diff
@@ -0,0 +1,1521 @@
+        let Pool {
+            authority: _,
+            start_time: _,
+            end_time: _,
+            reward_remaining: _,
+            stake,
+            reward,
+            last_update_time: _,
+            acc_reward_per_share: _,  // accumulated Coin<R> per share, times REWARD_PER_SHARE_MUL.
+        } = removed;
+
+        coin::destroy_zero(stake);
+
+        // There may be a single AU of reward remaining
+        if (coin::value(&reward) > 0) {
+            coin::deposit(sender_addr, reward);
+        } else {
+            coin::destroy_zero(reward);
+        };
+    }
+
+    /// Modify the incentive pool authority.
+    /// Only the pool's current authority can modify the authority.
+    public entry fun modify_authority<S, R>(
+        sender: &signer,
+        id: u64,
+        new_authority: address
+    ) acquires Pools {
+        assert!(exists<Pools<S, R>>(@aux), E_INCENTIVE_POOL_NOT_FOUND);
+        let pools = borrow_global_mut<Pools<S, R>>(@aux);
+        assert!(table::contains(&mut pools.pools, id), E_INCENTIVE_POOL_NOT_FOUND);
+        let pool = table::borrow_mut(&mut pools.pools, id);
+
+        let sender_addr = signer::address_of(sender);
+        assert!(sender_addr == pool.authority, E_NOT_AUTHORIZED);
+        let now = timestamp::now_microseconds();
+        update_pool(pool, now);
+
+        pool.authority = new_authority;
+
+        event::emit_event<ModifyPoolEvent<S, R>>(
+            &mut pools.modify_pool_events,
+            ModifyPoolEvent {
+                pool_id: id,
+                authority: pool.authority,
+                start_time: pool.start_time,
+                end_time: pool.end_time,
+                reward_remaining: pool.reward_remaining,
+                total_amount_staked: coin::value(&pool.stake),
+                acc_reward_per_share: pool.acc_reward_per_share
+            }
+        );
+    }
+
+    /// Modify incentive pool parameters.
+    /// Only the pool's authority can modify parameters.
+    public entry fun modify_pool<S, R>(
+        sender: &signer,
+        id: u64,
+        reward_amount: u64,
+        reward_increase: bool,
+        time_amount_us: u64,
+        time_increase: bool,
+    ) acquires Pools {
+        assert!(exists<Pools<S, R>>(@aux), E_INCENTIVE_POOL_NOT_FOUND);
+        let pools = borrow_global_mut<Pools<S, R>>(@aux);
+        assert!(table::contains(&mut pools.pools, id), E_INCENTIVE_POOL_NOT_FOUND);
+        let pool = table::borrow_mut(&mut pools.pools, id);
+
+        let sender_addr = signer::address_of(sender);
+        assert!(sender_addr == pool.authority, E_NOT_AUTHORIZED);
+
+        let now = timestamp::now_microseconds();
+        update_pool(pool, now);
+
+        if (reward_amount > 0 && reward_increase) {
+            let reward = coin::withdraw<R>(sender, reward_amount);
+            coin::merge(&mut pool.reward, reward);
+            pool.reward_remaining = pool.reward_remaining + reward_amount;
+            // TODO
```

Comment:
You mentioned that you wanted to put something here, but I can't recall what it was.

Reply:
ah, yep it's actually resolved i'll remove that